lib/inspec/dependencies/dependency_set.rb

```diff
@@ -26,7 +26,7 @@ def self.from_array(dependencies, cwd, cache, backend)
       dep_list = {}
       dependencies.each do |d|
         # if depedent profile does not have a source version then only name is used in dependency hash
-        key_name = (d.source_version.blank? ? "#{d.name}" : "#{d.name}-#{d.source_version}") rescue "#{d.name}"
```

Comment:
Having check both for `empty` and `nil` should work, since `blank?` checks for both.